app/src/main/java/com/github/k409/fitflow/FitFlowApplication.kt

```diff
@@ -45,67 +74,99 @@ class FitFlowApplication : Application(), Configuration.Provider, ImageLoaderFac
+            ExistingPeriodicWorkPolicy.CANCEL_AND_REENQUEUE,
+            drinkWorkerRequest,
+        )
     }
 
-    private inline fun <reified T : ListenableWorker> scheduleWork(workerName: String, repeatInterval: Long, timeUnit: TimeUnit, initialDelay: Long = 0L) {
+    private inline fun <reified T : ListenableWorker> scheduleWork(
+        workerName: String,
+        repeatInterval: Long,
+        timeUnit: TimeUnit,
+        initialDelay: Long = 0L,
+    ) {
         val workRequest = PeriodicWorkRequestBuilder<T>(repeatInterval, timeUnit)
             .apply {
                 if (initialDelay > 0) {
                     setInitialDelay(initialDelay, TimeUnit.MILLISECONDS)
                 }
             }.build()
 
-        WorkManager.getInstance(this).enqueueUniquePeriodicWork(
+        workManager.enqueueUniquePeriodicWork(
             workerName,
             ExistingPeriodicWorkPolicy.UPDATE,
             workRequest,
         )
     }
 
     private fun calculateInitialDelayUntilMidnight(): Long {
-        val calendar = Calendar.getInstance()
-        val now = calendar.timeInMillis
-
-        calendar.add(Calendar.DAY_OF_YEAR, 1)
-        calendar.set(Calendar.HOUR_OF_DAY, 0)
-        calendar.set(Calendar.MINUTE, 0)
-        calendar.set(Calendar.SECOND, 0)
-        calendar.set(Calendar.MILLISECOND, 0)
+        val now = LocalDateTime.now()
+        val nextMidnight = LocalDateTime.of(now.toLocalDate().plusDays(1), LocalTime.MIDNIGHT)
 
-        return calendar.timeInMillis - now
+        return Duration.between(now, nextMidnight).toMillis()
     }
 
     private fun calculateInitialDelayBeforeMidnight(): Long {
-        val calendar = Calendar.getInstance()
-        val now = calendar.timeInMillis
-
-        calendar.add(Calendar.DAY_OF_YEAR, 1)
-        calendar.set(Calendar.HOUR_OF_DAY, 23)
-        calendar.set(Calendar.MINUTE, 58)
-        calendar.set(Calendar.SECOND, 0)
-        calendar.set(Calendar.MILLISECOND, 0)
+        val now = LocalDateTime.now()
+        val beforeMidnight = LocalDateTime.of(now.toLocalDate().plusDays(1), LocalTime.of(23, 58))
 
-        return calendar.timeInMillis - now
+        return Duration.between(now, beforeMidnight).toMillis()
     }
 
-    override val workManagerConfiguration: Configuration
-        get() = Configuration.Builder()
-            .setMinimumLoggingLevel(android.util.Log.DEBUG)
-            .setWorkerFactory(workerFactory)
-            .build()
+    private fun createNotificationChannels() {
+        val manager = NotificationManagerCompat.from(this)
+
+        NotificationChannel.entries.forEach {
+            if (it != null) {
+                val notificationChannel = android.app.NotificationChannel(
+                    it.channelId,
+                    it.channelId,
+                    NotificationManager.IMPORTANCE_DEFAULT,
+                )
+
+                manager.createNotificationChannel(notificationChannel)
+            }
+        }
```

Comment:
Condition 'it != null' is always 'true'

Reply:
There is/was bug in Kotlin (don't know if fixed) where enum not initialize correctly and could return null